stl/inc/random

```diff
@@ -1974,8 +1972,13 @@ private:
             _Nx  = (_Wx + _Mx - 1) / _Mx + _Nfix; // trial _Nx
             _Wx0 = _Wx / _Nx;
             _Nx0 = _Nx - _Wx % _Nx;
-            _Yx0 = (_Rx >> _Wx0) << _Wx0;
-            _Yx1 = (((_Rx >> _Wx0) >> 1) << _Wx0) << 1;
+            if (_Wx0 < sizeof(_Eres) * CHAR_BIT) {
+                _Yx0 = (_Rx >> _Wx0) << _Wx0;
+                _Yx1 = (((_Rx >> _Wx0) >> 1) << _Wx0) << 1;
+            } else {
+                _Yx0 = 0;
+                _Yx1 = 0;
+            }
             if (_Nfix == 1 || _Rx - _Yx0 <= _Yx0 / _Nx) {

```

Comment:
Style nitpick: We conventionally place a newline between non-chained `if`-statements.